src/main/scala/extractor/SemanticCtx.scala

```diff
@@ -132,24 +133,26 @@ case class SemanticCtx(database: Database, projectPath: AbsolutePath) {
 }
 
 object SemanticDBFileVisitor {
-  def apply[T](filePath: Path, projectPath: AbsolutePath, results: ConcurrentLinkedQueue[(String, T)], f: SemanticCtx => Iterable[(String, T)]) = {
+  def apply[T](filePath: Path, f: SemanticCtx => Result): Result = {
       try {
         val sdb = s.Database.parseFrom(Files.readAllBytes(filePath))
         val mdb = sdb.toDb(None)
-        val ctx = SemanticCtx(mdb, projectPath)
-        f(ctx).foreach(results.add)
+        val ctx = SemanticCtx(mdb)
+        val res = f(ctx)
         print(".")
```

Comment:
I think it is better to add the `Logging` trait and add a real `logger.debug(s"Processing $filePath")` message